### ISSUE TITLE
Articulations (choose a macro set)

### ISSUE THREAD

**Issue by j5v** [surge-synthesizer/surge#6974]
This feature seems likely to exist in other synths, and someone more experienced than me could cut through this long description and just point to its name :)

**Is your feature request related to a problem? Please describe!**
As a sound designer, encouraged to make patches with macros, I'd like to make the patches more useful to users, by directing them to different useful 'sweet spots' that the patch offers. Users would then more quickly be able to explore the possibilities of a patch, or reset macros to preset/useful values.

**Describe the solution you'd like:**
If Surge offered **Articulations** (or some other name) in a menu, then users could select those to bulk-select macro values, or even store their own within the patch. Ideally (and by default anyway?) these macro changes are recorded in the DAW, if armed.

For example, as a designer, I could list out several piano configurations, like 'lively', 'felt', 'filtered', 'hammers', 'bass', 'ambient', and so on.

The UI might be challenging: Selection (surface UI), and then (deeper menu) Add, Delete, Rename for Articulations, and [stretch goal] Edit, to toggle whether each macro is changed by selection of the articulation.

More than macros?: not for MVP. Setting, say, parameters, is effectively a macro and wouldn't be transparent for the user; the answer for this use case is "more macros".

**Describe alternatives you've considered:**
It's possible to save them as separate patches, at the cost of fragmenting the different variants.

**Additional context:**
Con: As a designer, it's more challenging to hold all the articulations together, than it is to design specialist patches.
Con: If designers had included Articulations, and then change the main patch parameters, then it's a burden to recalibrate the Articulations.
Con: If a patch is able to cross categories with different Articulations, then it's hard to categorize (as a designer) and hard to find (as a user). For example, if a Keys instrument could be made to sound 'bowed', then users won't find the bowed variant easily because the main patch is in Keys.

**Comment by baconpaul** [comment]
Oh without committing to doing this any time soon, let me say I really like this idea! I think it would also make macros even more useful, which is definitely one of our goals.

**Comment by baconpaul** [comment]
basically internally we would have, like
```
articulations: [
   "soft": {0.2, -0.4, 0.5, ...},
  "hard", {0.9, 0.0, 0.2 ...}
]
```

just a list of named lists of values. Yeah?

**Comment by baconpaul** [comment]
If we did want this in a 1.x lifetime, the obvious UI would be

1. drop the "LIST" vertical button a bit
2. Have a hamburger menu next to the macro modulators
3. add a "save articulation as" there and list the articulations there

**Comment by j5v** [comment]
> just a list of named lists of values. Yeah?

That's the MVP, yes :)

I can see a use case where only some macros would want to be overwritten (bit mask, or Boolean-per-macro), but it's getting a bit edge-case. The MVP would be great as-is.

**Comment by j5v** [comment]
> obvious UI would be

4. A 'x' icon next to each articulation, to remove it, like we do for modulation targets/sources on RMB.

**Comment by baconpaul** [comment]
Or even just a little overlay 
I more wanted to make sure I understood the idea
It’s super interesting 
And also let’s you consider the alchemy trick of morph between articulations with midi and stuff

**Comment by mkruselj** [comment]
I would call them snapshots since it's a more generic term.

Not a bad idea but I feel we will be hard pressed to find people to even fill just the 8 macros with useful non boring assignments, let alone save multiple snapshots of them. Kinda feel you'd maybe be the only one using it, John. 😛

**Comment by j5v** [comment]
> Kinda feel you'd maybe be the only one using it, John. 😛
Yeah, I get that feeling too, with the tenuous hope that others might.
It looks like other plugins just use dedicated patches. I'm happy for this just to be parked.

**Comment by j5v** [comment]
> And also let’s you consider the alchemy trick of morph between articulations with midi and stuff
I'd thought that too. It's nice to think about, but it would get technically specialized with mappers, and would likely better suit an architecture that had macros onto a modular graph. As ED says, I think we're getting into the '<1% of users' territory.

**Comment by mkruselj** [comment]
> I'm happy for this just to be parked.

Let's do that then. :)
